docs/reference/google_drive.md

```diff
@@ -36,12 +39,29 @@ To get started, log into [Google Cloud Platform](cloud.google.com) and go to the
 
   Note: When you grant a service account access to a specific folder or shared drive in Google Drive, it's important to note that the permissions extend to all the children within that folder or drive. This means that any folders or files contained within the granted folder or drive inherit the same access privileges as the parent.
 
+
+#### Additional authentication prerequisites for document-level security
+
+To access user data on a Google Workspace domain, the service account that you created needs to be granted access by a super administrator for the domain. You can follow [official documentation](https://developers.google.com/cloud-search/docs/guides/delegation) to perform Google Workspace domain-wide delegation of authority.
+
+You need to grant following **OAuth Scopes** to your service account:
+- `https://www.googleapis.com/auth/admin.directory.group.readonly`
+- `https://www.googleapis.com/auth/admin.directory.user.readonly`
+
+This step allows the connector to access user data and their group memberships in your Google Workspace organization.
+
 ## Configuration
 
 The following configuration fields need to be provided for setting up the connector:
 ##### `Google Drive service account JSON`
 The service account credentials generated from Google Cloud Platform (JSON string). Refer to the [Google Cloud documentation](https://developers.google.com/workspace/guides/create-credentials#create_credentials_for_a_service_account) for more information.
 
+##### `Enable document level security`
```

Comment:
```suggestion
##### `Enable document-level security`
```
(Nit)
(just for consistency, optionally change all `document-level` to `document level`)